kafka-rest/src/main/java/io/confluent/kafkarest/response/StreamingResponse.java

```diff
@@ -79,6 +84,16 @@
               new V3ExceptionMapper(),
               response -> ((ErrorResponse) response.getEntity()).getErrorCode(),
               response -> ((ErrorResponse) response.getEntity()).getMessage())
+          .putMapper(
+              RestConstraintViolationException.class,
+              new RestConstraintViolationExceptionMapper(),
+              response -> ((ErrorResponse) response.getEntity()).getErrorCode(),
+              response -> ((ErrorResponse) response.getEntity()).getMessage())
+          .putMapper(
+              IllegalArgumentException.class,
+              new IllegalArgumentExceptionMapper(),
+              response -> Status.BAD_REQUEST.getStatusCode(),
+              response -> ((ErrorResponse) response.getEntity()).getMessage())
```

Comment:
This looks way too generic - can't we handle it more specifically? There's a chance we might end up masking some genuine programming errors as bad requests. I know we're manually using the mapper, but it still looks like too wide of a net.

Reply:
I've taken it out - I can't get it to break without the mapper in place, as we catch the IllegalArgumentException in ProduceAction anyway, around the getSchema call.

It could be argued that even there the catch is too generic, but the alternative is to put a try catch around each/all of the checkArugments in getSchema, and that feels a bit long winded.

I think not having it in the exception mapper, and having it around the getSchema call in the produceAction is probably the right balance.